service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -142,12 +152,57 @@ private boolean registerSchemas() {
       boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
       LOGGER.info("schemaId {} exists {}", schemaId, exists);
       if (!exists) {
-        if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
-          return false;
+        if (serviceCenterEnv()) {
+          if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
+            return false;
+          }
+        }
+      } else {
+        //查询是否存在schema,如果不存在则注册
+        String serviceId = microservice.getServiceId();
+        String schema = srClient.getSchema(serviceId, schemaId);
+        if (null == schema) {
+          LOGGER.warn("schemacontent {}/{} not exist.", serviceId, schemaId);
+          if (!srClient.registerSchema(serviceId, schemaId, content)) {
+            return false;
+          }
+        } else {
+          //查询是否存在摘要,不存在则注册
+          String schemaSummary = srClient.getSchemaSummary(serviceId, schemaId);
+          if (null == schemaSummary) {
+            LOGGER.warn("schemasummary {}/{} not exist.", serviceId, schemaId);
+            if (!srClient.registerSchema(serviceId, schemaId, content)) {
+              return false;
+            }
+          } else {
+            //比较本地摘要,如果不匹配,查询SC环境
+            String localSummary = Sha.getSHA256StrJava(content);
+            if (!localSummary.equals(schemaSummary)) {
+              LOGGER.warn("localsummary is not equal to what is already exist in servicecenter");
+              if (serviceCenterEnv()) {
+                if (!srClient.registerSchema(serviceId, schemaId, content)) {
+                  return false;
+                }
+              }
+            }
+          }
         }
       }
     }
 
     return true;
   }
+
+  private boolean serviceCenterEnv() {
+    if (Const.SERVICECENTER_RUNMODE_DEV.equals(srClient.getServiceCenter().getRunMode())) {
+      LOGGER.warn("The current servicecenter environment runmode is {}", Const.SERVICECENTER_RUNMODE_DEV);
+      return true;
+    } else if (Const.SERVICECENTER_RUNMODE_PROD.equals(srClient.getServiceCenter().getRunMode())) {
```

Comment:
already return, no need else

Reply:
ok，i'll modify it to return false directly,without using else